src/main/java/uk/gov/register/store/postgres/BatchedPostgresDataAccessLayer.java

```diff
@@ -168,11 +168,20 @@ public void addItem(Item item) {
     }
 
     @Override
-    public Optional<Item> getItem(HashValue hash) {
+    public Optional<Item> getItemByOldHash(HashValue hash) {
         if (batchedItems.containsKey(hash)) {
             return Optional.of(batchedItems.get(hash));
         }
         
+        return isEntireRegisterBatched ? Optional.empty() : postgresDataAccessLayer.getItemByOldHash(hash);
+    }
+
+    @Override
+    public Optional<Item> getItem(HashValue hash) {
+        // TODO: this layer uses old hashes to store stuff in memory, so this lookup
+        // can't be cached. If this is caching is still necessary, migrate this to use
+        // new hashes.
```

Comment:
I will look at this if/when we implement v2 of the `/load-rsf` endpoint. In the meantime this layer doesn't need to do anything because this method will only be called once per API request.